x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -554,6 +561,16 @@ public QueryBuilder semanticQuery(InferenceResults inferenceResults, float boost
                             );
                         }
 
+                        final Integer requestSize = searchExecutionContext.requestSize();
+                        if (Objects.equals(requestSize, 0)) {
+                            // Rewrite to a query that will return all docs with indexed embeddings for the queried field.
+                            // Size is often set to 0 when calculating aggregations and an unconstrained kNN query will return all docs
+                            // with indexed embeddings for the queried field, so we rewrite to a query that achieves that as efficiently as
+                            // possible.
+                            // TODO: Apply pre-filters once they are available
+                            yield new ExistsQueryBuilder(inferenceResultsFieldName);
```

Comment:
technically we wouldn't even need to do an inference call in this case? because it gets rewritten to an exists query either way. not saying we should necessarily do that here, happy to just fix the issue.

Reply:
Great point! The only edge case is that it is not rewritten to an exists query if the field uses a sparse vector model. If we could determine the embedding type on the coordinating node prior to inference, we could rewrite to exists there. I don't know if there's a currently a mechanism to get the model type without performing inference.